pyfrc/mains/cli_deploy.py

```diff
@@ -205,6 +209,32 @@ def run(self, options, robot_class, **static_options):
         print("\nSUCCESS: Deploy was successful!")
         return 0
 
+    def _generate_build_data(self, robot_path) -> dict:
+        """
+        Generate a deploy.json
+        """
+
+        hash = ""
+
+        try:
+            repoProcess = subprocess.run(
+                args=["git", "describe", "--dirty=-dirty", "--always"],
+                capture_output=True,
+            )
+            hash = repoProcess.stdout.decode().strip()
+        except subprocess.CalledProcessError as e:
+            logging.exception(e)
```

Comment:
This is going to be really obnoxious for anyone who is trying to deploy something that isn't in a git repo.

Reply:
thats a good idea, would be nice to too perhaps ommit the `hash` key rather than pass an empty string, or replace the string with a note like `not a repo`

Reply:
Actually, you should probably *first* check to see if you're in a git repo via https://stackoverflow.com/a/2180367/5483904 (and always ignore the error). Then if you are in a git repo you can run this command, and if there's an error you can display it to the user.